### PR TITLE
Tkp/contact importer

### PR DESCRIPTION
**Pull request by mrfelton**
Initial concept for a helper that automatically batches contacts into sets of 1500 and throttles the API usage to a maximum throughput of 3 requests per 2 seconds (the limit imposed by SendGrid). Results of each batch import are emitted as an event containing details of the batch (the contacts that were uploaded and the results of upload).

Addresses #273.

Note, this patch builds on work done in #275.

### CODE REVIEW

File: test/helpers/contact-importer/contact-importer.test.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      console.log('SUCCESS error', error)
+      console.log('SUCCESS batch', batch)
+    })
+    this.contactImporter.on('drain', function() {
+      expect(self.contactImporter._sendBatch).to.have.callCount(3)
+      done()
+    })
+
+
+    // assert.equal(sg.limiters['/v3/contactdb/recipients'].queue.length, 2)
```
Comment (beeman) [inline test/helpers/contact-importer/contact-importer.test.js]: @mrfelton commented code in test

**Comment by thinkingserious** [comment]
Hi @mrfelton,

Is this one ready for review to merge?

**Comment by mrfelton** [comment]
@thinkingserious yes, we're using this helper class in production. It could probably do with some more unit tests, but it does it's thing. I just successfully used it to import 199,713 contacts into a Sendgrid account.

**Comment by thinkingserious** [comment]
Hi @mrfelton,

Now that I've merged https://github.com/sendgrid/sendgrid-nodejs/pull/277, I was looking to merge this one. I noticed that there is some overlapping code though. I'm not sure if I should have not merged #277 in favor of this one. Could you please take a look and let me know what you think? Thanks!

**Comment by mrfelton** [comment]
@thinkingserious the changes in this PR were built ontop of the other one. It should apply cleanly ontop of it.

**Comment by thinkingserious** [comment]
Can you please send us your mailing address and T-shirt size to dx@sendgrid.com?

**Pull request merged**